middleware/getResults.js

```diff
@@ -0,0 +1,41 @@
+const http = require('http');
+const async = require('async');
+const xmlParser = require('../lib/xmlParser');
+const utils = require('../lib/resultsUtils');
+const processResults = require('../lib/resultsProcessor.js');
+
+function getData(requestUrl, page, cb) {
+  const pagedRequestUrl = `${requestUrl}${page}`;
+
+  http.get(pagedRequestUrl, (res) => {
+    let xml = '';
+
+    res.on('data', (chunk) => {
+      xml += chunk;
+    });
+
+    res.on('end', () => {
+      // TODO: Handle responses that are not 200
+      const pharmacyList = xmlParser(xml);
+      cb(null, pharmacyList);
+    });
+  });
+}
+
+function getResults(req, res, next) {
+  const requestUrl = utils.generateRequestUrl(req);
```

Comment:
Not much really, I was thinking about it but decided against it mainly for the reason of it feels less like a process the request needs to do through and more of a function something else relies on.